gcapi/gcapi.py

```diff
@@ -282,12 +379,17 @@ class AlgorithmResultsAPI(APIBase):
 class AlgorithmJobsAPI(APIBase):
     base_path = "algorithms/jobs/"
 
+    @filter_arguments(None, parse_uuid_from_url)
+    def by_input_image(self, pk):
+        return self.iterate_all(params={"image": pk})
```

Comment:
Wait, all this complexity is here just so you don't need to do `image.pk`? The `parse_uuid_from_url` is fragile and too much magic. The alternative would be on the grand challenge side to accept filters by api urls rather than pks, but not this.

Reply:
I just pretty much replicated what is already happening on the cirrus-core side. Ii made sense from my side to embed this in here instead.

`image.pk` is a bit too simple of an example though as there are many places where gcapi does return api_urls but no pks. Example: https://grand-challenge.org/api/v1/algorithms/e5d03359-4c31-46de-9dd6-e077d925dc4d/ - the jobs specifically do, too, but I do not have an example url handy.

However, if grand-challenge could ingest api_urls as alternative for pks that would be great, then we would not need this and can clean up the code in cirrus-core anyway! Certainly an alternative!